From fd44b82cc82df7b1e97a536f2b101c0eb08905d5 Mon Sep 17 00:00:00 2001
From: Yanteng Si <siyanteng@loongson.cn>
Date: Mon, 20 May 2024 10:20:21 +0800
Subject: [PATCH 114/360] PCI: fix kabi error caused by pm_suspend_target_state

fix kabi error caused by pm_suspend_target_state,used only
by loongson devices.

Signed-off-by: Hongchen Zhang <zhanghongchen@loongson.cn>
Signed-off-by: Yanteng Si <siyanteng@loongson.cn>
---
 drivers/pci/pci.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index e2abcfd67878..da31aa220d88 100644
--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -31,7 +31,9 @@
 #include <asm/dma.h>
 #include <linux/aer.h>
 #include <linux/bitfield.h>
+#ifdef CONFIG_MACH_LOONGSON64
 #include <linux/suspend.h>
+#endif
 #include "pci.h"
 
 DEFINE_MUTEX(pci_slot_mutex);
@@ -172,11 +174,15 @@ static bool pci_bridge_d3_disable;
 /* Force bridge_d3 for all PCIe ports */
 static bool pci_bridge_d3_force;
 
+#ifdef CONFIG_MACH_LOONGSON64
+
 #ifndef CONFIG_PM_SLEEP
 suspend_state_t pm_suspend_target_state;
 #define pm_suspend_target_state (PM_SUSPEND_ON)
 #endif
 
+#endif
+
 static int __init pcie_port_pm_setup(char *str)
 {
 	if (!strcmp(str, "off"))
@@ -6035,8 +6041,9 @@ int pcie_set_readrq(struct pci_dev *dev, int rq)
 {
 	u16 v;
 	int ret;
+#ifdef CONFIG_MACH_LOONGSON64
 	struct pci_host_bridge *bridge = pci_find_host_bridge(dev->bus);
-
+#endif
 	if (rq < 128 || rq > 4096 || !is_power_of_2(rq))
 		return -EINVAL;
 
@@ -6054,6 +6061,7 @@ int pcie_set_readrq(struct pci_dev *dev, int rq)
 
 	v = FIELD_PREP(PCI_EXP_DEVCTL_READRQ, ffs(rq) - 8);
 
+#ifdef CONFIG_MACH_LOONGSON64
 	if (pm_suspend_target_state == PM_SUSPEND_ON &&
 		bridge->no_inc_mrrs) {
 		int max_mrrs = pcie_get_readrq(dev);
@@ -6063,6 +6071,7 @@ int pcie_set_readrq(struct pci_dev *dev, int rq)
 			return -EINVAL;
 		}
 	}
+#endif
 
 	ret = pcie_capability_clear_and_set_word(dev, PCI_EXP_DEVCTL,
 						  PCI_EXP_DEVCTL_READRQ, v);
-- 
2.20.1

